include/mxnet/c_lapack_api.h

```diff
@@ -54,6 +57,40 @@ extern "C" {
 
 inline char loup(char uplo, bool invert) { return invert ? (uplo == 'U' ? 'L' : 'U') : uplo; }
 
+
+/*!
+ * \brief Transpose matrix data in memory
+ * 
+ * Equivalently we can see it as flipping the layout of the matrix
+ * between row-major and column-major.
+ *
+ * \param m number of rows of input matrix a
+ * \param n number of columns of input matrix a
+ * \param b output matrix
+ * \param ldb leading dimension of b
+ * \param a input matrix
+ * \param lda leading dimension of a
+ */
+template <typename xpu, typename DType>
+inline void flip(int m, int n, DType *b, int ldb, DType *a, int lda);
```

Comment:
This function is intended to be a helper function that just does "flipping" or transpose of elements in memory, for the use of other MXNet LAPACK interfaces. It sits beneath those interfaces.

By the design in `include/mxnet/c_lapack_api.h`, all MXNet LAPACK interfaces need to be able to handle both row-major and column-major matrices. When we're building against LAPACK libs that only support column-major matrices natively, we'll have to provide support for row-major ones. Sometimes we could do it cleverly as in `dgesdd` (SVD), but many more times we do have to transpose some input matrix under the hood before feeding to the native LAPACK routine, e.g. `dposv` (linear solver).

People may argue that for many functions, we could do things in a theoretically equivalent way without having to do hard transpose of input matrix. I did some test -- we may lose on numerical accuracy in the general case. It seems to me that when we do need it, transposing prior and after the native call seems an easy-to-implement, numerically conservative option.